src/components/income/OtherInvestments.tsx

```diff
@@ -0,0 +1,198 @@
+    return undefined
+  }
+  return {
+    name,
+    openDate,
+    closeDate,
+    openPrice: Number(openPrice),
+    closePrice: Number(closePrice),
+    quantity: Number(quantity),
+    state
+  }
+}
+
+export default function OtherInvestments(): ReactElement {
+  const assets = useSelector((state: YearsTaxesState) => state.assets)
+
+  const methods = useForm<AssetUserInput>()
+  const { handleSubmit, watch } = methods
+  const positionType = watch('positionType')
+  const dispatch = useDispatch()
+
+  const { onAdvance, navButtons } = usePager()
+
+  const onSubmitAdd = (formData: AssetUserInput): void => {
+    const payload = toAsset(formData)
+    if (payload !== undefined) {
+      dispatch(addAsset(payload))
+    }
+  }
+
+  const onSubmitEdit =
+    (index: number) =>
+    (formData: AssetUserInput): void => {
+      const payload = toAsset(formData)
+      if (payload !== undefined) {
+        dispatch(editAsset({ value: payload, index }))
+      }
+    }
+
+  const form: ReactElement | undefined = (
+    <FormListContainer<AssetUserInput>
+      onSubmitAdd={onSubmitAdd}
+      onSubmitEdit={onSubmitEdit}
+      items={assets.map((a) => toUserInput(a))}
+      removeItem={(i) => dispatch(removeAsset(i))}
+      primary={(f) => f.name}
+      secondary={(f) =>
+        `${f.openDate?.toDateString() ?? ''}: ${f.quantity} @ $${f.openPrice}`
+      }
+    >
+      {' '}
+      <Grid container spacing={2}>
+        <p>Manage Assets</p>
+        <GenericLabeledDropdown<AssetType>
+          label="Asset Type"
+          name="positionType"
+          dropDownData={['Security', 'Real Estate']}
+          keyMapping={(x) => x}
+          textMapping={show(showAssetType)}
+          valueMapping={(x) => x}
+        />
+        <LabeledInput
+          label="Address"
+          patternConfig={Patterns.name}
+          name="name"
+        />
+        <DatePicker
+          maxDate={new Date(2021, 12, 31)}
+          label={positionType === 'Real Estate' ? 'Purchase date' : 'Open Date'}
+          name="openDate"
+        />
+        <DatePicker
+          maxDate={new Date(2021, 12, 31)}
+          label={positionType === 'Real Estate' ? 'Sale date' : 'Close Date'}
+          name="closeDate"
+        />
+        <LabeledInput
+          label="Cost of acquisition"
+          patternConfig={Patterns.currency}
+          name="openPrice"
+        />
+        {(() => {
+          if (positionType === 'Real Estate') {
+            return (
+              <>
+                <LabeledInput
+                  label="Net proceeds from sale"
+                  patternConfig={Patterns.currency}
+                  name="closePrice"
+                />
+                <USStateDropDown label="Property state" name="state" />
+              </>
+            )
+          } else {
+            return (
+              <>
+                <LabeledInput
+                  label="Quantity"
+                  patternConfig={Patterns.currency}
```

Comment:
I don't think Quantity is a currency type? Also, since we're concerning ourselves with quantity we should clarify that the cost of acquisition and sell price are either cost per unit or total amount of the given quantity.

Reply:
Yeah I'm thinking ahead to where lots can be split. So if I buy 100 units of x and then later sell 50 units, I need to subdivide the initial purchase block...might be confusing. Fixed format in cc8ecec